enhancements/proposals/perftune.md

```diff
@@ -0,0 +1,220 @@
+Because NodeTuner Pod is running on the same node where tuned Scylla is, so instance metadata will be correct.
+
+Currently only `--write-back-cache=false"` is required on GKE. Other Clouds use script without any additional parameters.
+
+#### IRQ Layout
+When any Scylla Pod is running on the host, IRQs are going to be pinned to CPUs not used by Scylla.
+
+In case when the last Scylla Pod is removed from the k8s node, perftune is going to set IRQ to all available CPUs, but it’s not going to cancel other optimizations done by the script.   
+In order to restore them, k8s node has to be restarted. Later we can think about fixing that.
+
+When irqbalancer is running on the host, script is banning CPUs assigned to handle IRQ inside irqbalancer config file, and initiates a restart.  
+We need a shared Host PID namespace for that.
+
+### CPU assignment discovery
+
+Only single supported Cloud Provider (EKS) exposes cpuset via Container Runtime Interface. 
+ScyllaNodeConfig Pod is going to first check if Scylla container cpuset is available in CRI. If not, it will fallback to 
+manually reading it from cgroup virtual filesystem.
+
+#### RBAC
+
+The DaemonSet is going to use separate ServiceAccount with ClusterRole with read permissions for Pods and write permissions for Events and ConfigMaps, Jobs.
+RBAC resources will be reconciled by the Operator.
+
+#### Selectors and ownership
+
+ScyllaNodeConfig DaemonSet
+
+```
+Labels: {
+  "scylla-operator.scylladb.com/node-config": snc.Name,
+}
+OwnerReferences: []metav1.OwnerReference{util.NewNodeConfigControllerRef(snc)},
+```    
+
+Perftune result ConfigMap:
+```
+Labels: {
+  "scylla-operator.scylladb.com/node-config":           snc.Name,
+  "scylla-operator.scylladb.com/node-config-controller" : sncPod.Name,
+},
+OwnerReferences: []metav1.OwnerReference{util.NewNodeConfigControllerRef(snc)},
+}
+```
+
+Perftune Job:
+```
+Labels: map[string]string{
+  "scylla-operator.scylladb.com/node-config":           snc.Name,
+  "scylla-operator.scylladb.com/node-config-controller": sncPod.Name,
+},
+OwnerReferences: []metav1.OwnerReference{util.NewNodeConfigControllerRef(snc)},
+```
+
+### Test Plan
+
+We are not going to test if each individual optimization was applied to the host machine, but instead we are going to 
+observe if a corresponding ConfigMap is created for each Scylla Pod.
+Node Affinity/Tolerations are going to be tested in unit tests. 
+E2E are going to cover ConfigMaps and DaemonSet reconcilation.
+
+### Upgrade / Downgrade Strategy
+
+DaemonSet definition will be reconciled by the Operator and annotated with hash of definition. 
+Hash annotation will make sure that newer/older version of DaemonSet is deployed. 
+
+### Version Skew Strategy
+
+For the short period of time DaemonSet and Operator version may differ. Because these components doesn't talk to each other directly, version mismatch doesn't matter.
```

Comment:
```suggestion
For a short period of time DaemonSet and Operator versions may differ. Because these components don't talk to each other directly, version mismatch doesn't matter.
```